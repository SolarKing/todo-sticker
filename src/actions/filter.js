// @flow

export type FilterActions =
    | DueDateFilterAction;

export type DueDateFilterAction = {|
    type: 'dueDateFilter',
    filter: string,
|}

export function settingDueDateFilter(dueDateFilter: string): DueDateFilterAction {
    return {
        type: 'dueDateFilter',
        filter: dueDateFilter
    };
}
