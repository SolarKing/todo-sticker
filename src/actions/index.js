// @flow
import type { TodoActions } from './todo';
import type { FilterActions } from './filter';
import type { SortActions } from './sort';

export type Actions =
    | TodoActions
    | FilterActions
    | SortActions;
