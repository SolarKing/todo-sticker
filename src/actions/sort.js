// @flow
import type {SortByType, SortColumn} from '../constant/sort';

export type SortActions =
    | SortByAction
    | SortColumnAction;

export type SortColumnAction = {|
    type: 'sortColumn',
    column: SortColumn
|};

export type SortByAction = {|
    type: 'sortBy',
    sortBy: SortByType
|};

export function settingSortColumn(column: SortColumn): SortColumnAction {
    return {
        type: 'sortColumn',
        column: column
    };
}
