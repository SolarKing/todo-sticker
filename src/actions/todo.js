// @flow
import type { TodoState } from '../states';

export type TodoActions = AddAction
    | EditAction
    | DeleteAction
    | CompleteAction;

export const TodoActionTypeEnum = {
    Add: 'add',
    Edit: 'edit',
    Delete: 'delete',
    Complete: 'complete'
};

export type AddAction = {|
    type: 'add',
    todoState: TodoState
|}

export type EditAction = {|
    type: 'edit',
    todoState: TodoState
|}

export type DeleteAction = {|
    type: 'delete',
    id: string
|}

export type CompleteAction = {|
    type: 'complete',
    id: string
|}

export function addTodo(todoState: TodoState): AddAction {
    return {
        type: TodoActionTypeEnum.Add,
        todoState
    };
}

export function editTodo(todoState: TodoState): EditAction {
    return {
        type: TodoActionTypeEnum.Edit,
        todoState
    };
}

export function deleteTodo(id: string): DeleteAction {
    return {
        type: TodoActionTypeEnum.Delete,
        id
    };
}

export function completeTodo(id: string): CompleteAction {
    return {
        type: TodoActionTypeEnum.Complete,
        id
    };
}
