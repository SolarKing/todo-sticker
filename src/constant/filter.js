// @flow
export type DueDateFilter = 'today' | 'all';

export const DueDateFilterEnum = {
    today: 'today',
    all: 'all'
};
