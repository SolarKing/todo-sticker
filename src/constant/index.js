// @flow

const DueDate = {
    today: 'today',
    all: 'all'
};


export const FilterEnum = {
    DueDate
};
