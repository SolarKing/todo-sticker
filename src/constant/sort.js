// @flow
export type SortColumn = 'dueDate' | 'createTime';
export type SortByType = 'desc' | 'asc';

export const SortColumnEnum = {
    dueDate: 'dueDate',
    createTime: 'createTime'
};

export const SortByTypeEnum = {
    desc: 'desc',
    asc: 'asc'
};
