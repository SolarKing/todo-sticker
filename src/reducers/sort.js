// @flow
import type {SortState} from '../states/index';
import {SortByTypeEnum, SortColumnEnum} from '../constant/sort';
import type {SortActions} from '../actions/sort';

const initState: SortState = {
    sortBy: SortByTypeEnum.asc,
    sortColumn: SortColumnEnum.dueDate
};

export default function sort(sortState: SortState = initState, action: SortActions) {
    switch (action.type) {
    case 'sortColumn':
        return {
            ...sortState,
            sortColumn: action.column
        };
    default:
        return sortState;
    }
}
