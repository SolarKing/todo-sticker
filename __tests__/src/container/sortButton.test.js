// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import SortButton from '../../../src/container/sortButton';
import { SortColumnEnum, SortByTypeEnum } from '../../../src/constant/sort';
import { Dropdown } from 'antd';
import configureStore from 'redux-mock-store';
import { settingSortColumn, settingSortBy } from '../../../src/actions/sort';

describe('SortButton', () => {
    let wrapper: ShallowWrapper;
    let wrapperContainer: ShallowWrapper;
    let store = configureStore()({
        sort: {
            sortBy: SortByTypeEnum.asc,
            sortColumn: SortColumnEnum.createTime
        }
    });

    beforeAll(() => {
        wrapperContainer = shallow(
            <SortButton
                store={store}
            />
        );

        wrapper = wrapperContainer.shallow();
    });

    afterEach(() => {
        store.clearActions();
    });

    it('should render correct as container', () => {
        expect(wrapperContainer).toMatchSnapshot();
    });

    it('should render correct as component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle sort column change', () => {
        let menuClickValue = {
            key : SortColumnEnum.dueDate,
        };
        let menu = shallow(wrapper.find(Dropdown).prop('overlay'));
        menu.simulate('click', menuClickValue);
        expect(store.getActions()).toEqual([settingSortColumn(SortColumnEnum.dueDate)]);
    });

    it('should handle sort rule change', () => {
        wrapper.find('SortRuleSpan').simulate('click');
        expect(store.getActions()).toEqual([settingSortBy(SortByTypeEnum.desc)]);
        
    });

});