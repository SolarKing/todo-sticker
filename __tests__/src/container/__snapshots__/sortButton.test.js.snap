// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButton should render correct as component 1`] = `
<div
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
>
  <p
    style={
      Object {
        "display": "inline",
      }
    }
  >
    Sort by
  </p>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <Menu
        className=""
        onClick={[Function]}
        prefixCls="ant-menu"
        theme="light"
      >
        <MenuItem>
          <a>
            Due Date
          </a>
        </MenuItem>
        <MenuItem>
          <a>
            Create Time
          </a>
        </MenuItem>
      </Menu>
    }
    placement="bottomLeft"
    prefixCls="ant-dropdown"
    trigger={
      Array [
        "click",
      ]
    }
  >
    <a>
      createTime
    </a>
  </Dropdown>
  <SortRuleSpan
    className="glyphicon glyphicon-sort-by-order"
    onClick={[Function]}
  />
</div>
`;

exports[`SortButton should render correct as container 1`] = `
<SortButton
  onSortByChanged={[Function]}
  onSortColumnSelected={[Function]}
  sortByType="asc"
  sortColumn="createTime"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
